Source/WebKit/WebProcess/WebPage/WebPage.cpp

```diff
@@ -6612,6 +6612,14 @@ void WebPage::drawPagesForPrinting(FrameIdentifier frameID, const PrintInfo& pri
 
 void WebPage::addResourceRequest(WebCore::ResourceLoaderIdentifier identifier, const WebCore::ResourceRequest& request)
 {
+    ASSERT(!m_networkResourceRequestIdentifiersForPageLoadTiming.contains(identifier));
+    {
+        bool wasEmpty = m_networkResourceRequestIdentifiersForPageLoadTiming.isEmpty();
+        m_networkResourceRequestIdentifiersForPageLoadTiming.add(identifier);
+        if (wasEmpty)
+            send(Messages::WebPageProxy::StartNetworkRequestsForPageLoadTiming());
```

Comment:
This function already sends Messages::WebPageProxy::SetNetworkRequestsInProgress below.  Could they be combined into one?

Reply:
No, the exact condition under which these messages are sent are different.